src/conv/heuristics/ai_heuristics.cpp

```diff
@@ -0,0 +1,445 @@
+            static_cast<float>(isFwd ? conv_problem.GetInWidth() : conv_problem.GetOutWidth()),
+            static_cast<float>(conv_problem.GetWeightsDepth()),
+            static_cast<float>(conv_problem.GetWeightsHeight()),
+            static_cast<float>(conv_problem.GetWeightsWidth()),
+            static_cast<float>(isFwd ? conv_problem.GetOutChannels()
+                                     : conv_problem.GetInChannels()),
+            static_cast<float>(isFwd ? conv_problem.GetOutDepth() : conv_problem.GetInDepth()),
+            static_cast<float>(isFwd ? conv_problem.GetOutHeight() : conv_problem.GetInHeight()),
+            static_cast<float>(isFwd ? conv_problem.GetOutWidth() : conv_problem.GetInWidth()),
+            static_cast<float>(conv_problem.GetOutBatchSize()),
+            static_cast<float>(1), // TunaNet was trained on a dataset of 2D
+                                   // problems where PadD was incorrectly set to 1
+            static_cast<float>(conv_problem.GetPadH()),
+            static_cast<float>(conv_problem.GetPadW()),
+            static_cast<float>(1), // TunaNet was trained on a dataset of 2D
+                                   // problems where StrideD was incorrectly set to 1
+            static_cast<float>(conv_problem.GetKernelStrideH()),
+            static_cast<float>(conv_problem.GetKernelStrideW()),
+            static_cast<float>(conv_problem.GetDilationH()),
+            static_cast<float>(conv_problem.GetDilationW()),
+            static_cast<float>(metadata.EncodeLayout(conv_problem.GetInLayout())),
+            static_cast<float>(metadata.EncodePrecision(conv_problem.GetInDataType())),
+            static_cast<float>(metadata.EncodeDirection(conv_problem.GetDirection())),
+            static_cast<float>(conv_problem.GetGroupCount())};
+
+        // normalize
+        for(size_t i = 0; i < features.size(); ++i)
+            features[i] = (features[i] - metadata.features_mean[i]) / metadata.features_std[i];
+
+        return features;
+    }
+};
+
+std::unique_ptr<Model> GetModel(const std::string&) { return std::make_unique<Gfx908Model>(); }
+
+std::vector<uint64_t> PredictSolver(const ProblemDescription& problem,
+                                    const ConvolutionContext& ctx,
+                                    const std::string& device)
+{
+    const static std::unique_ptr<Model> model = GetModel(device);
+    if(!model || !model->IsProblemSupported(problem, ctx))
+        return {};
+
+    std::string est_name = ":memory:" + device;
+    auto& db             = AnyRamDb::GetCached(est_name);
+    auto db_res          = db.FindRecord(problem.conv_problem);
+    if(db_res)
+    {
+        MIOPEN_LOG_I2("Cached heuristic result found");
+        std::vector<uint64_t> db_sol(db_res->size());
+        // cast returned record to solver ids
+        std::transform(db_res->begin(), db_res->end(), db_sol.begin(), [](boost::any id) {
+            return boost::any_cast<uint64_t>(id);
```

Comment:
Lets remove `boost::any` in favor of `std::any`

Reply:
This would require changes to the FindDb structs, which is more than the scope of this PR